Fix redis connection loop

Rotorsoft opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
The server application fails to load properly when the Redis service is unavailable. The bootstrapping function in `server.ts` is causing the application to become unhealthy in such scenarios.

## Link to Issue
Closes: #TODO

## Description of Changes
- Fixes redis adapter to ignore connection errors, and keep retrying until the app is disposed
- Removed max retries condition and caps retry timeout to 10 mins
- Sets connection timeout to 3 secs
- Dispose disconnects and quits redis
- initialized gate checks for isReady instead of isOpen
- Simplified adapter construction

timolegros commented on libs/adapters/src/redis/RedisCacheAdapter.ts:
```diff
@@ -385,14 +382,14 @@ export class RedisCache implements Cache {
     const data = {} as any;
     try {
       if (this.initialized()) {
```
Let's reverse all of these `if (this.initialized())` and remove the undefined return types

timolegros commented on libs/adapters/src/redis/RedisCacheAdapter.ts:
```diff
@@ -385,14 +382,14 @@ export class RedisCache implements Cache {
         }
         for (const key of keys) {
-          data[key] = await this._client.get(key);
+          data[key] = await this._client?.get(key);
```
Why `?` instead of `!` here?

timolegros reviewed (changes requested):
Looks good!

Suggested some additional clean-ups.

**Note**
JTN since we delay the first health ping we likely need to update the monitor in Datadog.

timolegros commented:
An additional change: if the `REDIS_URL` variable is undefined we should report the error once and don't try to reconnect. This will avoid cluttering the local logs when you don't have Redis running locally.

mhagel commented on packages/commonwealth/test/devnet/cosmos/cosmosGovNotifGenerator.spec.ts:
```diff
@@ -50,12 +56,34 @@ async function enableChains(chains: string[]) {
+          default_symbol: 'STAKE',
+          chain_node_id: chainNode.id,
+        },
+      });
```
@Rotorsoft I got this closer to the data we need, but something is still not right. @timolegros is more familiar with these particular tests, so maybe he can help.

Rotorsoft commented:
> An additional change: if the `REDIS_URL` variable is undefined we should report the error once and don't try to reconnect. This will avoid cluttering the local logs when you don't have Redis running locally.

@timolegros based on this comment, changed the strategy a little bit. Basically the redis adapter is only injected if REDIS_URL is defined, otherwise the default in-memory cache is used (NO-OP cache). Also modified CacheDecorator to use the cache() port. Resolved all your comments above by always instantiating this._client in the constructor, since now we are committing to inject redis when REDIS_URL is defined.

Rotorsoft commented:
@timolegros can you please review the failing `packages/commonwealth/test/devnet/cosmos/cosmosGovNotifGenerator.spec.ts` ? ... skipping until resolved. @mhagel thinks this might be a bug in the tests, and it prevents the other cosmos tests to pass.
